### PR TITLE
RDCC-992: Addressing QA Peer Review Comments: Profile Sync

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-992

### Change description ###

- [x] 1.  | Structure Test Packages To Match Main Packages: https://github.com/hmcts/rd-profile-sync/pull/181/commits/95fc7dac9ee2b394ceebef52e9e03b20c4ee33f1
- [x] 2.  | Re-organise Classes Into Correct Packages: https://github.com/hmcts/rd-profile-sync/pull/181/commits/be536750b77bfd620a65fcae49283af8bf22142f
- [x] 3.  | Optimise Imports: https://github.com/hmcts/rd-profile-sync/pull/181/commits/ff76b1e2d0eb93667773ddcbe86a025d2db44d01
- [x] 4.  | Amend Mocks Used In Test Classes: https://github.com/hmcts/rd-profile-sync/pull/181/commits/5482551e3976ef4d96a1c521807ece11bd4b8502
- [x] 5.  | Amend Test Class Names: _Not required, all tests end in **Test** already_
- [x] 6.  | Removing Unused Variables/Format Lines: _evidence:_ https://tinyurl.com/RDCC-1000
- [x] 7.  | Amend Tests To Include Verify: https://github.com/hmcts/rd-profile-sync/pull/181/commits/5482551e3976ef4d96a1c521807ece11bd4b8502
- [x] 8.  | Create Missing Tests For Untested Classes: https://github.com/hmcts/rd-profile-sync/pull/181/commits/0993fecdd13681d7ec06863105b24eeb1c69c0a0, https://github.com/hmcts/rd-profile-sync/pull/181/commits/5482551e3976ef4d96a1c521807ece11bd4b8502
- [x] 9.  | Create Tests for Repository Objects: https://github.com/hmcts/rd-profile-sync/pull/181/commits/4c4debcbc74b0060fba0712365778551064d7cbe
- [x] 10. | Increase Test Code Coverage to 80% : _evidence:_ https://tinyurl.com/RDCC-1004

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[✘] No
```

### CODE REVIEW

File: config/owasp/suppressions.xml [hunk 1]
```diff
@@ -2,6 +2,49 @@
...
+    </suppress>
+    <suppress>
+        <notes><![CDATA[
+   This suppresses a CVE-2020-5529 cve in any htmlunit of selenium
+   ]]></notes>
+        <cpe>cpe:/a:htmlunit:htmlunit</cpe>
+        <cve>CVE-2020-5529</cve>
+    </suppress>
+    <suppress>
+        <notes><![CDATA[
```
Comment (AdilOoze) [inline config/owasp/suppressions.xml]: These Suppressions are already in Master, moving them here to resolve dependency-check failure in Jenkins

File: src/test/java/uk/gov/hmcts/reform/profilesync/domain/SyncJobConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.profilesync.domain;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+public class SyncJobConfigTest {
+
+    String firstSearchQuery = "firstsearchquery";
+    String configRun = "1h";
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/profilesync/domain/SyncJobConfigTest.java]: private final

File: src/test/java/uk/gov/hmcts/reform/profilesync/domain/SyncJobAuditTest.java [hunk 3]
```diff
@@ -3,32 +3,35 @@
...
 import java.time.LocalDateTime;
+
 import org.junit.Test;
+import uk.gov.hmcts.reform.profilesync.constants.Source;
 
-public class SyncJobTest {
+public class SyncJobAuditTest {
 
+    private String success = "success";
+    private String errorMsg = "errorMsg";
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/profilesync/domain/SyncJobAuditTest.java]: add final keyword

File: src/test/java/uk/gov/hmcts/reform/profilesync/helper/MockDataProvider.java [hunk 4]
```diff
@@ -8,8 +8,8 @@
 import java.util.List;
 
 import uk.gov.hmcts.reform.profilesync.client.IdamClient;
-import uk.gov.hmcts.reform.profilesync.domain.GetUserProfileResponse;
 import uk.gov.hmcts.reform.profilesync.domain.UserProfile;
+import uk.gov.hmcts.reform.profilesync.domain.response.GetUserProfileResponse;
 
 public class MockDataProvider {
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/profilesync/helper/MockDataProvider.java]: if the variable  declaration is static final make  it in the capital letters .
example public static final String idamId  
change idamId to IDAMID

File: src/test/java/uk/gov/hmcts/reform/profilesync/util/UserProfileSyncJobSchedulerTest.java [hunk 5]
```diff
@@ -2,81 +2,84 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/profilesync/util/UserProfileSyncJobSchedulerTest.java]: move it schedular package

File: build.gradle [hunk 6]
```diff
@@ -58,7 +58,7 @@ jacocoTestCoverageVerification {
 			limit {
 				counter = 'LINE'
 				value = 'COVEREDRATIO'
```
[the remark this answers is not in the document]
Comment (AdilOoze) [reply build.gradle]: Same as Master, however I will increase to 80

File: build.gradle [hunk 7]
```diff
@@ -168,7 +168,7 @@ pitest {
 	threads = 10
 	outputFormats = ['XML', 'HTML']
 	timestampedReports = false
-	mutationThreshold = 80
```
[the remark this answers is not in the document]
Comment (AdilOoze) [reply build.gradle]: Same as Master, however I will increase to 80

File: src/test/java/uk/gov/hmcts/reform/profilesync/helper/MockDataProvider.java (hunk 4 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/profilesync/helper/MockDataProvider.java]: Will make it how all constants are with underscores:  _IDAM_ID_

File: src/test/java/uk/gov/hmcts/reform/profilesync/util/UserProfileSyncJobSchedulerTest.java (hunk 5 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/profilesync/util/UserProfileSyncJobSchedulerTest.java]: this has been addressed in the following commit: https://github.com/hmcts/rd-profile-sync/pull/181/commits/bc37603637e5fcbbc46bdd8621f47415d846e2c2

File: src/test/java/uk/gov/hmcts/reform/profilesync/helper/MockDataProvider.java (hunk 4 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/profilesync/helper/MockDataProvider.java]: this has been addressed in the following commit: https://github.com/hmcts/rd-profile-sync/pull/181/commits/bc37603637e5fcbbc46bdd8621f47415d846e2c2

File: src/test/java/uk/gov/hmcts/reform/profilesync/domain/SyncJobAuditTest.java (hunk 3 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/profilesync/domain/SyncJobAuditTest.java]: this has been addressed in the following commit: https://github.com/hmcts/rd-profile-sync/pull/181/commits/bc37603637e5fcbbc46bdd8621f47415d846e2c2

File: src/test/java/uk/gov/hmcts/reform/profilesync/domain/SyncJobConfigTest.java (hunk 2 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/profilesync/domain/SyncJobConfigTest.java]: this has been addressed in the following commit: https://github.com/hmcts/rd-profile-sync/pull/181/commits/bc37603637e5fcbbc46bdd8621f47415d846e2c2

File: src/main/java/uk/gov/hmcts/reform/profilesync/util/UserProfileSyncJobScheduler.java [hunk 8]
```diff
@@ -10,10 +10,10 @@
 import org.springframework.beans.factory.annotation.Value;
```
[the remark this answers is not in the document]
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/profilesync/util/UserProfileSyncJobScheduler.java]: this has been addressed in the following commit: https://github.com/hmcts/rd-profile-sync/pull/181/commits/bc37603637e5fcbbc46bdd8621f47415d846e2c2

File: build.gradle (hunk 7 above)
[the remark this answers is not in the document]
Comment (AdilOoze) [reply build.gradle]: this has been addressed in the following commit: https://github.com/hmcts/rd-profile-sync/pull/181/commits/bc37603637e5fcbbc46bdd8621f47415d846e2c2

File: build.gradle (hunk 6 above)
[the remark this answers is not in the document]
Comment (AdilOoze) [reply build.gradle]: this has been addressed in the following commit: https://github.com/hmcts/rd-profile-sync/pull/181/commits/bc37603637e5fcbbc46bdd8621f47415d846e2c2

File: build.gradle [hunk 9]
```diff
@@ -12,14 +12,14 @@ plugins {
...
 	id "info.solidsoft.pitest" version '1.4.5'
-	id 'io.spring.dependency-management' version '1.0.8.RELEASE'
-	id 'org.owasp.dependencycheck' version '5.2.4'
+	id 'io.spring.dependency-management' version '1.0.9.RELEASE'
+	id 'org.owasp.dependencycheck' version '5.3.0'
 	id 'org.sonarqube' version '2.8'
-	id 'org.springframework.boot' version '2.1.4.RELEASE'
-	id "org.flywaydb.flyway" version "6.1.1"
-	id 'au.com.dius.pact' version '4.0.3'
+	id 'org.springframework.boot' version '2.2.4.RELEASE'
```
Comment (shreedhar-hmcts) [inline build.gradle]: any reason why the dependency versions downgraded ?

Comment (AdilOoze) [reply build.gradle]: It seems Profile Sync's Demo branch did not have latest versions

Comment (shreedhar-hmcts) [reply build.gradle]: but why you are downgrading the versions of dependencies?

Comment (AdilOoze) [reply build.gradle]: This has been addressed in the following commits: https://github.com/hmcts/rd-profile-sync/pull/181/commits/5338b31bf1246beb5f9d4302a78f4830dc0e4372 & https://github.com/hmcts/rd-profile-sync/pull/181/commits/8cf58e15e02afb1a2f6452cb589eacc6bff12bb3

**Comment by shashank-rastogi** [comment]
👍  for the Change description!!

**Pull request merged**